### PR TITLE
Adding syncmaster&worker reconciliation support.

### PR DESCRIPTION
**Pull request by ewoutp**
Several sync related fixes, tests & improvements.

This PR includes #143

### CODE REVIEW

**Comment by matthewvon** [comment]
WAIT ... pushed approved on wrong PR ... just starting on this one.

File: pkg/apis/deployment/v1alpha/sync_spec_test.go [hunk 1]
```diff
@@ -27,39 +27,31 @@ import (
...
 	assert.False(t, def(SyncSpec{}).IsEnabled())
 	assert.False(t, def(SyncSpec{Enabled: util.NewBool(false)}).IsEnabled())
 	assert.True(t, def(SyncSpec{Enabled: util.NewBool(true)}).IsEnabled())
-	assert.Equal(t, "test-image", def(SyncSpec{}).GetImage())
-	assert.Equal(t, "foo", def(SyncSpec{Image: util.NewString("foo")}).GetImage())
-	assert.Equal(t, v1.PullAlways, def(SyncSpec{}).GetImagePullPolicy())
-	assert.Equal(t, v1.PullNever, def(SyncSpec{ImagePullPolicy: util.NewPullPolicy(v1.PullNever)}).GetImagePullPolicy())
 	assert.Equal(t, "test-jwt", def(SyncSpec{}).Authentication.GetJWTSecretName())
 	assert.Equal(t, "foo", def(SyncSpec{Authentication: SyncAuthenticationSpec{JWTSecretName: util.NewString("foo")}}).Authentication.GetJWTSecretName())
 }
```
Comment (matthewvon) [inline pkg/apis/deployment/v1alpha/sync_spec_test.go]: no test for new parameter to monitoring.SetDefaults("string")

Comment (ewoutp) [reply pkg/apis/deployment/v1alpha/sync_spec_test.go]: Good! Added some tests

**Comment by ewoutp** [comment]
Unit test failures have been fixed

**Pull request merged**